src/smif/data_layer/memory_interface.py

```diff
@@ -363,10 +363,20 @@ def write_state(self, state, modelrun_name, timestep=None, decision_iteration=No
 
     # region Conversion coefficients
     def read_coefficients(self, source_spec, destination_spec):
-        return self._coefficients[(source_spec, destination_spec)]
+        spec = (source_spec.name, destination_spec.name)
+        if spec not in self._coefficients.keys():
```

Comment:
Style nit: `if spec not in self._coefficients` would be more idiomatic than `.keys()` - though `try ... except KeyError` might be better than `if ... else` here anyway.

Reply:
Fixed